### PR TITLE
Gist Shortcode: refactor unit tests

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

* Gist shortcode: refactor shortcode tests
* Gist embeds: refactor tests to use single data provider (and remove the unnecessary `do_action( 'init' )` calls)
* Remove file from exclude list now that it is lint free

#### Jetpack product discussion

* p9dueE-3LL-p2

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Do the tests pass?

### CODE REVIEW

File: projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.gist.php [hunk 1]
```diff
@@ -40,595 +49,237 @@ public function test_shortcodes_gist() {
...
-	 */
-	public function test_shortcodes_gist_oembed_to_embed() {
+	public function test_gist_embeds( $content, $expected, $expected_amp ) {
 		global $post;
 
-		$gist_id = '57cc50246aab776e110060926a2face2';
-		$url     = 'https://gist.github.com/' . $gist_id;
-		$post    = $this->factory()->post->create_and_get( array( 'post_content' => $url ) );
-
-		do_action( 'init' );
```
Comment (jeherve) [inline projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.gist.php:450]: This is removed and not added back.

File: projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.gist.php [hunk 2]
```diff
@@ -40,595 +49,237 @@ public function test_shortcodes_gist() {
...
-	 */
-	public function test_shortcodes_gist_file_to_embed() {
-		global $post;
-
-		$gist_id = 'jeherve/57cc50246aab776e110060926a2face2';
-		$file    = 'wp-config-php';
-		$url     = 'https://gist.github.com/' . $gist_id . '#file-' . $file;
-		$post    = $this->factory()->post->create_and_get( array( 'post_content' => $url ) );
-
-		do_action( 'init' );
```
Comment (jeherve) [inline projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.gist.php:485]: This is removed and not added back.

File: projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.gist.php [hunk 3]
```diff
@@ -40,595 +49,237 @@ public function test_shortcodes_gist() {
...
+	 * @covers ::github_gist_shortcode
 	 */
-	public function test_shortcodes_gist_oembed_with_tab_size() {
-		global $post;
-
-		$gist_id = '57cc50246aab776e110060926a2face2';
-		$url     = 'https://gist.github.com/' . $gist_id . '/?ts=4';
-		$post    = $this->factory()->post->create_and_get( array( 'post_content' => $url ) );
-
-		do_action( 'init' );
```
Comment (jeherve) [inline projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.gist.php:615]: This is removed and not added back.

**Review by obenland — APPROVED** [review]
Thank you, it fixed the fatal when running PHPUnit on dotcom! The 19 failing tests there are for another time.

**Review by sdixon194 — APPROVED** [review]

**Comment by kraftbj** [comment]
r235364-wpcom

**Pull request merged**